smartapps/smartthings/ecobee-connect.src/ecobee-connect.groovy

```diff
@@ -422,10 +461,11 @@ def initialize() {
 	def childThermostats = thermostats.collect { dni ->
 		def d = getChildDevice(dni)
 		if(!d) {
-			d = addChildDevice(app.namespace, getChildName(), dni, null, ["label":"${thermostatList[dni]}" ?: getChildName()])
+			d = addChildDevice(app.namespace, getChildName(), dni, null, ["label":"${thermostatList[dni].data.name}" ?: getChildName()])
```

Comment:
should we add a null safety check for `thermostatList[dni]?.data?.name` just in case?

Reply:
name comes from getThermostatDisplayName(data) where data is resp.data.thermostatList, if name isn't present in that data it will default to (stat.modelNumber == "siSmart" ? "Smart Si" : "Smart") + " (${stat.identifier})").toString(), so there should always be a name present, if the name still is empty it will default to "Ecobee Thermostat"